elixir/APIRequest/lib/APIRequest.ex

```diff
@@ -0,0 +1,151 @@
+#
+# This file is part of AtomVM.
+#
+# Copyright 2018 Davide Bettio <davide@uninstall.it>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+#
+
+defmodule APIRequest do
+  def start() do
+    :io.format(~c"starting API Request application~n")
+    spawn(fn -> start_network() end)
+
+    case wait_for_network() do
+      :ok ->
+        :io.format(~c"Connected to WiFi! Making HTTP request...~n")
+
+      {:error, reason} ->
+        :io.format(~c"Failed to connect to WiFi: ~p~n", [reason])
+    end
+  end
+
+  defp start_network() do
+    :io.format(~c"Starting network...~n")
+
+    config = [
```

Comment:
You could choose to make this a demo, but if you would like to keep it as a simple example instead, I recommend using [`: wait_for_sta/1` (or /2)](https://www.atomvm.net/doc/main/apidocs/erlang/eavmlib/network.html#wait-for-sta-1) and not use the callbacks. The goal for the examples is to keep them to a very limited scope. If you would like to submit a separate PR with an Elixir example for connecting to WiFi and using the callbacks (like the Erlang [wifi](https://github.com/atomvm/atomvm_examples/tree/master/erlang/wifi) example) that would be a very welcome addition too!